server/src/main/kotlin/suwayomi/tachidesk/manga/impl/backup/proto/ProtoBackupImport.kt

```diff
@@ -313,128 +291,108 @@ object ProtoBackupImport : ProtoBackupBase() {
-        val dbChapters = ChapterTable.selectAll().where { ChapterTable.manga eq mangaId }
+        ChapterTable.batchInsert(chaptersToInsert) { chapter ->
+            this[ChapterTable.url] = chapter.url
+            this[ChapterTable.name] = chapter.name
+            if (chapter.date_upload == 0L) {
+                this[ChapterTable.date_upload] = chapter.date_fetch
+            } else {
+                this[ChapterTable.date_upload] = chapter.date_upload
+            }
+            this[ChapterTable.chapter_number] = chapter.chapter_number
+            this[ChapterTable.scanlator] = chapter.scanlator
 
-        uniqueChapters.forEach { chapter ->
-            val dbChapter = dbChapters.find { it[ChapterTable.url] == chapter.url }
+            this[ChapterTable.sourceOrder] = chaptersToInsert.size - chapter.source_order
+            this[ChapterTable.manga] = mangaId
 
-            if (dbChapter == null) {
-                ChapterTable.insert {
-                    it[url] = chapter.url
-                    it[name] = chapter.name
-                    if (chapter.date_upload == 0L) {
-                        it[date_upload] = chapter.date_fetch
-                    } else {
-                        it[date_upload] = chapter.date_upload
-                    }
-                    it[chapter_number] = chapter.chapter_number
-                    it[scanlator] = chapter.scanlator
+            this[ChapterTable.isRead] = chapter.read
+            this[ChapterTable.lastPageRead] = chapter.last_page_read.coerceAtLeast(0)
+            this[ChapterTable.isBookmarked] = chapter.bookmark
 
-                    it[sourceOrder] = chaptersLength - chapter.source_order
-                    it[ChapterTable.manga] = mangaId
+            this[ChapterTable.fetchedAt] = TimeUnit.MILLISECONDS.toSeconds(chapter.date_fetch)
+        }
 
-                    it[isRead] = chapter.read
-                    it[lastPageRead] = chapter.last_page_read.coerceAtLeast(0)
-                    it[isBookmarked] = chapter.bookmark
-                }
-            } else {
-                ChapterTable.update({ (ChapterTable.url eq dbChapter[ChapterTable.url]) and (ChapterTable.manga eq mangaId) }) {
-                    it[isRead] = chapter.read || dbChapter[isRead]
-                    it[lastPageRead] = max(chapter.last_page_read, dbChapter[lastPageRead]).coerceAtLeast(0)
-                    it[isBookmarked] = chapter.bookmark || dbChapter[isBookmarked]
+        if (chaptersToUpdateToDbChapter.isNotEmpty()) {
+            BatchUpdateStatement(ChapterTable).apply {
+                chaptersToUpdateToDbChapter.forEach { (backupChapter, dbChapter) ->
+                    addBatch(EntityID(dbChapter[ChapterTable.id].value, ChapterTable))
+                    this[ChapterTable.isRead] = backupChapter.read || dbChapter[ChapterTable.isRead]
+                    this[ChapterTable.lastPageRead] = max(backupChapter.last_page_read, dbChapter[ChapterTable.lastPageRead])
```

Comment:
```suggestion
                    this[ChapterTable.lastPageRead] = max(backupChapter.last_page_read, dbChapter[ChapterTable.lastPageRead]).coerceAtLeast(0)
```
Forgot this already 😄

Reply:
![tenor](https://github.com/user-attachments/assets/7b6191bd-32f3-4cd6-8c26-c4764285659d)